### PR TITLE
Add doc/parameters for RF Regression and LR

### PR DESCRIPTION
**Pull request by wbo4958**

### CODE REVIEW

**Comment by wbo4958** [comment]
build

File: python/src/spark_rapids_ml/regression.py [hunk 1]
```diff
@@ -257,9 +301,27 @@ class LinearRegression(
 
     """
 
-    def __init__(self, **kwargs: Any) -> None:
+    @keyword_only
+    def __init__(
+        self,
+        *,
+        featuresCol: Union[str, List[str]] = "features",
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py:308]: I can't find a way to add these constructor args automatically. Yeah, let's make it the ground truth source.

File: python/src/spark_rapids_ml/regression.py [hunk 2]
```diff
@@ -573,10 +635,82 @@ class RandomForestRegressor(
...
     :py:class:`cuml.ensemble.RandomForestRegressor`. And it can automatically map
     pyspark parameters to cuML parameters.
 
+
+    Parameters
+    ----------
+
+    featuresCol:
+        The feature column names, spark-rapids-ml supports vector, array and columnar as the input.\n
+            * When the value is string, the feature columns must be assembled into 1 column with vector or array type.
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py]: Done

File: python/src/spark_rapids_ml/regression.py [hunk 3]
```diff
@@ -573,10 +635,82 @@ class RandomForestRegressor(
...
+            * When the value is a list of string, the feature columns must be numeric types.
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py]: Done

File: python/src/spark_rapids_ml/regression.py [hunk 4]
```diff
@@ -573,10 +635,82 @@ class RandomForestRegressor(
...
+    labelCol:
+        The label column name.
+    predictionCol:
+        The predictionCol column name.
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py]: Done

File: python/src/spark_rapids_ml/regression.py [hunk 5]
```diff
@@ -573,10 +635,82 @@ class RandomForestRegressor(
...
+            * ``0`` - Disables all log messages.
+            * ``1`` - Enables only critical messages.
+            * ``2`` - Enables all messages up to and including errors.
+            * ``3`` - Enables all messages up to and including warnings.
+            * ``4 or False`` - Enables all messages up to and including information messages.
+            * ``5 or True`` - Enables all messages up to and including debug messages.
+            * ``6`` - Enables all messages up to and including trace messages.
+    n_streams:
+        Number of parallel streams used for forest building.
+        Please note that there is a bug running spark-rapids-ml on the node with multi-gpus
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py]: Done

File: python/src/spark_rapids_ml/regression.py [hunk 6]
```diff
@@ -202,6 +202,50 @@ class LinearRegression(
...
+    ----------
+
+    featuresCol:
+        The feature column names, spark-rapids-ml supports vector, array and columnar as the input.\n
+            * When the value is string, the feature columns must be assembled into 1 column with vector or array type.
+            * When the value is a list of string, the feature columns must be numeric types.
+    labelCol:
+        The label column name.
+    predictionCol:
+        The predictionCol column name.
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py]: Done

File: python/src/spark_rapids_ml/regression.py [hunk 7]
```diff
@@ -202,6 +202,50 @@ class LinearRegression(
         cases only if features and labels are standardized in the input dataframe.  Otherwise,
         they will match only if regParam = 0 or elastNetParam = 1.0 (aka Lasso).
 
+    Parameters
+    ----------
+
+    featuresCol:
+        The feature column names, spark-rapids-ml supports vector, array and columnar as the input.\n
+            * When the value is string, the feature columns must be assembled into 1 column with vector or array type.
+            * When the value is a list of string, the feature columns must be numeric types.
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py]: Done

File: python/src/spark_rapids_ml/regression.py [hunk 8]
```diff
@@ -202,6 +202,50 @@ class LinearRegression(
         cases only if features and labels are standardized in the input dataframe.  Otherwise,
         they will match only if regParam = 0 or elastNetParam = 1.0 (aka Lasso).
 
+    Parameters
+    ----------
+
+    featuresCol:
+        The feature column names, spark-rapids-ml supports vector, array and columnar as the input.\n
+            * When the value is string, the feature columns must be assembled into 1 column with vector or array type.
```
[the remark this answers is not in the document]
Comment (wbo4958) [reply python/src/spark_rapids_ml/regression.py]: Done

**Comment by wbo4958** [comment]
build

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**